libraries/chain/include/sophiatx/chain/genesis_state.hpp

```diff
@@ -31,47 +31,154 @@
 #include <string>
 #include <vector>
 
-namespace sophiatx { namespace chain {
+namespace sophiatx {
+namespace chain {
 using std::string;
 using std::vector;
 using namespace sophiatx::protocol;
 
 struct genesis_state_type {
 
-   struct initial_account_type {
-      initial_account_type(const account_name_type& name = string(),
-                           const public_key_type& key = public_key_type(),
-                           const share_type& balance = 0
-      )
-            : name(name),
-              key(key),
-              balance(balance)
-      {}
-      account_name_type name;
-      public_key_type key;
-      share_type balance;
-   };
-
-   public_key_type                          initial_public_key = SOPHIATX_INIT_PUBLIC_KEY;
-   share_type                               initial_balace = SOPHIATX_INIT_SUPPLY;
-
-   vector<initial_account_type>             initial_accounts;
-
-
-   chain_id_type                            initial_chain_id;
-   time_point_sec                           genesis_time;
-   bool                                     is_private_net = false;
-
-   /**
-    * Get the chain_id corresponding to this genesis state.
-    *
-    * This is the SHA256 serialization of the genesis_state.
-    */
-   chain_id_type compute_chain_id() const{return initial_chain_id;};
+    struct initial_account_type {
+        initial_account_type(const account_name_type &name = string(),
+                             const public_key_type &key = public_key_type(),
+                             const share_type &balance = 0
+        )
+                : name(name),
+                  key(key),
+                  balance(balance) {}
+
+        account_name_type name;
+        public_key_type key;
+        share_type balance;
+    };
+
+    public_key_type initial_public_key = sophiatx::protocol::public_key_type(SOPHIATX_INIT_PUBLIC_KEY_STR);
+
+    public_key_type initial_public_mining_key = sophiatx::protocol::public_key_type(SOPHIATX_INIT_PUBLIC_KEY_STR);
+#undef SOPHIATX_INIT_PUBLIC_KEY_STR
+
```

Comment:
might be nicer to use default value directly here, no #define and #undef is then needed

Reply:
Yes, I agree, but then there will be some defines in config.hpp and some in here, so it is more confusing

Reply:
how about making all the newly added parameters in genesis optional ? Then genesis state does not have to have default values(default genesis.txt stays the same as now or can optionally include those new params), we can delete all the macros and default values will be in init function(or constructor) of config class and only those genesis parameters that have some value will be overridden in config